File: CsLox/com/craftinginterpreters/lox/LoxDictGet.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace com.craftinginterpreters.lox
{
    internal class LoxDictGet : LoxCallable
    {
        readonly LoxDict dict;

        public LoxDictGet(LoxDict dict)
        {
            this.dict = dict;
        }

        public int arity()
        {
            return 1;
        }

        public Object call(Interpreter interpreter, List<Object> arguments)
        {
            Object key = (Object)arguments[0];
            return dict.elements[key];
        }
    }
}

File: JavaLox/src/com/craftinginterpreters/lox/LoxDictGet.java
package com.craftinginterpreters.lox;

import java.util.List;

class LoxDictGet implements LoxCallable {
   final LoxDict dict;
   
   public LoxDictGet(LoxDict dict) {
      this.dict = dict;
   }
   
   @Override
   public int arity() {
      return 1;
   }
   
   @Override
   public Object call(Interpreter interpreter, List<Object> arguments) {
      Object key = arguments.get(0);
      return dict.elements.get(key);
   }
}
